tmt/steps/provision/__init__.py

```diff
@@ -317,31 +365,43 @@ class GuestFacts(SerializableContainer):
         default=cast(Optional['tmt.package_managers.GuestPackageManager'], None))
 
     has_selinux: Optional[bool] = None
+    has_systemd: Optional[bool] = None
     is_superuser: Optional[bool] = None
     is_ostree: Optional[bool] = None
 
-    #: Various Linux capabilities and whether they are permitted to
-    #: commands executed on this guest.
-    capabilities: dict[GuestCapability, bool] = field(
-        default_factory=cast(Callable[[], dict[GuestCapability, bool]], dict),
-        serialize=lambda capabilities: {
-            capability.value: enabled
-            for capability, enabled in capabilities.items()
-            } if capabilities else {},
-        unserialize=lambda raw_value: {
-            GuestCapability(raw_capability): enabled
-            for raw_capability, enabled in raw_value.items()
-            }
-        )
+    #: Permitted Linux capabilities on this guest.
+    capabilities: Optional[list[GuestCapability]] = field(
+        default=None,
+        serialize=lambda caps: [cap.value for cap in caps] if caps else None,
+        unserialize=lambda serialized: [GuestCapability(value) for value in serialized]
+        if serialized else None)
 
     os_release_content: dict[str, str] = field(default_factory=dict)
     lsb_release_content: dict[str, str] = field(default_factory=dict)
 
     def has_capability(self, cap: GuestCapability) -> bool:
+        """
+        Check if the guest has a specific capability.
+
+        :param cap: The capability to check for.
+        :returns: True if the guest has the specified capability, False otherwise.
+        """
+        if not self.capabilities or cap not in self.capabilities:
+            return False
+
+        return True
+
+    def has_capabilities(self, caps: list[GuestCapability]) -> bool:
```

Comment:
All we need is an iterable, so `*caps: GuestCapability` would give us a tuple, and we could call this method without constructing a list, `has_capability(GuestCapability.CAP_SYSLOG, GuestCapability.CAP_PERFMON)`.

And then you can drop the `has_capability` method because it would be the same as `has_capabilities(a_single_capability)` :)

Reply:
Addressed in 62ad828b667fc6680072f375e6bb30edff9725a9